defaults/defaults.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Program otelsvc is the Open Telemetry Service that collects stats
+// and traces and exports to a configured backend.
+
+package defaults
+
+import (
+	"github.com/open-telemetry/opentelemetry-service/exporter"
+	"github.com/open-telemetry/opentelemetry-service/exporter/loggingexporter"
+	"github.com/open-telemetry/opentelemetry-service/exporter/opencensusexporter"
+	"github.com/open-telemetry/opentelemetry-service/exporter/prometheusexporter"
+	"github.com/open-telemetry/opentelemetry-service/internal"
+	"github.com/open-telemetry/opentelemetry-service/internal/collector/processor/nodebatcher"
+	"github.com/open-telemetry/opentelemetry-service/internal/collector/processor/queued"
+	"github.com/open-telemetry/opentelemetry-service/processor"
+	"github.com/open-telemetry/opentelemetry-service/processor/addattributesprocessor"
+	"github.com/open-telemetry/opentelemetry-service/receiver"
+	"github.com/open-telemetry/opentelemetry-service/receiver/jaegerreceiver"
+	"github.com/open-telemetry/opentelemetry-service/receiver/opencensusreceiver"
+	"github.com/open-telemetry/opentelemetry-service/receiver/prometheusreceiver"
+	"github.com/open-telemetry/opentelemetry-service/receiver/vmmetricsreceiver"
+	"github.com/open-telemetry/opentelemetry-service/receiver/zipkinreceiver"
+)
+
+// Components returns the default set of components used by the
+// opentelemetry service
+func Components() (
+	map[string]receiver.Factory,
+	map[string]processor.Factory,
+	map[string]exporter.Factory,
+	error,
+) {
+	errs := []error{}
+	receivers, err := receiver.Factories(
```

Comment:
I think a rename here will be good: instead of `receiver.Factories` something like `receiver.BuildFactoryMap`, the first gives an impression of some global registry of factories, the second makes clear that it is just building the map.

Reply:
same apply to processors and exporters

Reply:
Makes sense. I think `factories.Build()` will be better. It's shorter but has the same meaning I think. Does that sound good?